src/app/content/components/Page/contentLinkHandler.ts

```diff
@@ -39,24 +42,28 @@ const reducePageReferenceError = (reference: PageReferenceError, document: Docum
 };
 
 // tslint:disable-next-line: max-line-length
-const reduceReference = (reference: PageReferenceMap, currentPath: string, document: Document, systemQueryString: string) => {
+const reduceReference = (reference: PageReferenceMap, currentPath: string, document: Document, systemQueryParams: SystemQueryParams) => {
   const path = content.getUrl(reference.params);
+  const options = createNavigationOptions(systemQueryParams);
   const a = assertNotNull(
     document.querySelector(`[href^='${reference.match}']`),
     'references are created from hrefs');
-  const href = assertNotNull(a.getAttribute('href'), 'it was found by href value')
-    .replace(reference.match, toRelativeUrl(currentPath, path) + systemQueryString);
-  a.setAttribute('href', href);
+  const href = assertNotNull(a.getAttribute('href'), 'it was found by href value');
+  // add hash to options if href only consists of hash, ie an anchor on the same page
+  options.hash = (a as HTMLAnchorElement).hash === href ? new URL(href, `https://openstax.org${currentPath}`).hash : '';
```

Comment:
i think its ok to unconditionally forward the hash, if its a link to a figure on another page we want that to work too

Reply:
i didn't realize the anchor element exposes `hash`, you probably don't need all that `URL` junk